### PR TITLE
a tiny js world completed

### PR DESCRIPTION
**Pull request by kotlyar-andrey**
### a tiny js world

[Demo](https://kotlyar-andrey.github.io/a-tiny-JS-world/) |
[Code base](https://github.com/kotlyar-andrey/a-tiny-JS-world)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

File: submissions/kotlyar-andrey/a-tiny-JS-world/index.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+  this.properties = [
+    "species",
+    "name",
+    "gender",
+    "legs",
+    "hands",
+    "saying",
+    "friendsNames",
+  ];
```
Comment (OleksiyRudenko) [inline submissions/kotlyar-andrey/a-tiny-JS-world/index.js:24]: Properties that aren't intended for direct external (outside class definition) access should be declared as private. This is achieved by prepending their names with `_` (conventional) or `#` (class syntax). The latter is not applicable here.

File: submissions/kotlyar-andrey/a-tiny-JS-world/index.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  this.saying = saying;
+
+  this.properties = [
+    "species",
+    "name",
+    "gender",
+    "legs",
+    "hands",
+    "saying",
+    "friendsNames",
```
Comment (OleksiyRudenko) [inline submissions/kotlyar-andrey/a-tiny-JS-world/index.js:23]: Cannot see `friendNames` among class properties. This confuses.

File: submissions/kotlyar-andrey/a-tiny-JS-world/index.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+const cat = new Creature("cat", "Jerry", "male", 4, 0, "Myau");
+const dog = new Creature("dog", "Bars", "male", 4, 0, "Woof");
+const woman = new Creature("human", "Maria", "female", 2, 2, "Hi");
+const man = new Creature("human", "Ivan", "male", 2, 2, "Hello");
+
+man.friends = [woman, cat, dog];
+woman.friends = [man, cat];
+dog.friends = [man];
```
Comment (OleksiyRudenko) [inline submissions/kotlyar-andrey/a-tiny-JS-world/index.js:43]: Creating new properties outside class definitions is a bad practice.

File: submissions/kotlyar-andrey/a-tiny-JS-world/index.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+const inhabitants = [man, woman, dog, cat];
+
+inhabitants.forEach((creature) => {
+  creature.display();
+});
```
Comment (OleksiyRudenko) [inline submissions/kotlyar-andrey/a-tiny-JS-world/index.js:49]: When an arrow function has a single parameter it is OK to skip parentheses.
If an arrow function body has a single statement it is OK to skip curly brackets.
Less brackets == better readability.
Note, that function without `return` statement still implies `return` statement (equivalent to `return undefined;`)

File: submissions/kotlyar-andrey/a-tiny-JS-world/index.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    "saying",
+    "friendsNames",
+  ];
+
+  this.display = function () {
+    if (this.friends) {
+      this.friendsNames = this.friends.map((friend) => friend.name).join(", ");
+    } else {
+      this.friendsNames = "undefined :(";
+    }
```
Comment (OleksiyRudenko) [inline submissions/kotlyar-andrey/a-tiny-JS-world/index.js:31]: How about ternary operator here?

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@kotlyar-andrey very well done!
A few improvements and we're good to go.

When improving (and actually initially writing your code), imagine
(a) someone else will be tasked to add new features or entities - we want to care about them and avoid confusions;
(b) codebase is big, i.e. logical components (e.g. class definition, objects instantiation, business logic) are split into individual files, number of objects is quite big, so that we cannot really rely on quick scroll to check things.

**Comment by kotlyar-andrey** [comment]
Thanks for the detailed feedback! This inspired me to change the code using a class. Please, review.

**Comment by OleksiyRudenko** [comment]
Does it work as intended?

image

**Comment by kotlyar-andrey** [comment]
Yes, it's my test message `cat.saying = "new message for test woman-cats";`
for testing

> whenever you change cat's saying cat-woman's saying should change accordingly, they are strongly tied on some astral level

**Review by OleksiyRudenko — APPROVED** [review]
@kotlyar-andrey Great job!

**Pull request merged**